File: Code Generation/responses/cpp/claude/medium/properties-graph.cpp
#include <vector>
#include <unordered_set>
using namespace std;

class Solution {
    vector<int> parent;
    int find(int x) { return parent[x] == x ? x : parent[x] = find(parent[x]); }
    void unite(int a, int b) { parent[find(a)] = find(b); }
public:
    int numberOfComponents(vector<vector<int>> properties, int k) {
        int n = properties.size();
        parent.resize(n);
        iota(parent.begin(), parent.end(), 0);
        
        vector<unordered_set<int>> sets(n);
        for (int i = 0; i < n; i++)
            for (int v : properties[i]) sets[i].insert(v);
        
        for (int i = 0; i < n; i++)
            for (int j = i+1; j < n; j++) {
                int cnt = 0;
                for (int v : sets[i]) if (sets[j].count(v)) cnt++;
                if (cnt >= k) unite(i, j);
            }
        
        int components = 0;
        for (int i = 0; i < n; i++) if (find(i) == i) components++;
        return components;
    }
};

File: Code Generation/responses/python/claude/medium/properties-graph.py
from typing import List

class Solution:
    def numberOfComponents(self, properties: List[List[int]], k: int) -> int:
        n = len(properties)
        sets = [set(p) for p in properties]
        parent = list(range(n))

        def find(x):
            while parent[x] != x:
                parent[x] = parent[parent[x]]
                x = parent[x]
            return x

        def union(x, y):
            parent[find(x)] = find(y)

        for i in range(n):
            for j in range(i + 1, n):
                if len(sets[i] & sets[j]) >= k:
                    union(i, j)

        return len({find(i) for i in range(n)})